### PR TITLE
Feature/FOUR-11572-11575 Draft

### PR DESCRIPTION
**Pull request by gustavobascope**
## Issue & Reproduction Steps
Improve Table's View

## Solution
- A new Table component was created

## How to Test
Requests option

## Related Tickets & Packages
https://processmaker.atlassian.net/browse/FOUR-11609
https://processmaker.atlassian.net/browse/FOUR-11614
https://processmaker.atlassian.net/browse/FOUR-11618
https://processmaker.atlassian.net/browse/FOUR-11633
https://processmaker.atlassian.net/browse/FOUR-11632
https://processmaker.atlassian.net/browse/FOUR-11637
https://processmaker.atlassian.net/browse/FOUR-11634
https://processmaker.atlassian.net/browse/FOUR-11617
https://processmaker.atlassian.net/browse/FOUR-11624
https://processmaker.atlassian.net/browse/FOUR-11639
https://processmaker.atlassian.net/browse/FOUR-11640
https://processmaker.atlassian.net/browse/FOUR-11631
https://processmaker.atlassian.net/browse/FOUR-11652
https://processmaker.atlassian.net/browse/FOUR-11653
https://processmaker.atlassian.net/browse/FOUR-11656

ci:deploy
ci:next

### CODE REVIEW

File: resources/js/components/PMColumnFilterPopover/PMColumnFilterForm.vue [hunk 1]
```diff
@@ -0,0 +1,164 @@
+<template>
+  <div class="pm-filter-form">
+    <b-form>
+      <b-form-group>
+        <b-button variant="light"
```
Comment (rodriquelca) [inline resources/js/components/PMColumnFilterPopover/PMColumnFilterForm.vue:5]: maybe onSortAscending and onSortDesending buttons could abstract as a dynamic component with custom params  (svg icon, title, click handler and toggle event)

File: resources/js/components/PMColumnFilterPopover/PMColumnFilterForm.vue [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    components: {
+    },
+    data() {
+      return {
+        toggleAsc: true,
+        toggleDesc: false,
+        items: []
+      };
+    },
+    created() {
```
Comment (rodriquelca) [inline resources/js/components/PMColumnFilterPopover/PMColumnFilterForm.vue:76]: remove empty method

File: resources/js/components/PMColumnFilterPopover/PMColumnFilterForm.vue [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        this.items = [];
+        this.addItem(0);
+      },
+      onChangeOperator(item, index) {
+        if (item.operator === '') {
+          item.subject = '';
+          item.logicalOp = '';
+          this.onChangeLogicalOp(item, index);
+        }
+        //let sw = item.operator !== '';
```
Comment (rodriquelca) [inline resources/js/components/PMColumnFilterPopover/PMColumnFilterForm.vue:104]: remove commented code

File: resources/js/components/PMColumnFilterPopover/PMColumnFilterForm.vue [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+          this.addItem(next);
+        }
+        if (item.logicalOp === '') {
+          item.logicalOp = this.items[next].logicalOp;
+          item._addedCondition = item.logicalOp !== '';
+          this.removeItem(next);
+        }
+      },
+      addItem(index) {
+        let item = {
```
Comment (rodriquelca) [inline resources/js/components/PMColumnFilterPopover/PMColumnFilterForm.vue:121]: item must be received as a method parameter

File: resources/js/components/PMColumnFilterPopover/PMColumnFilterForm.vue [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+          operator: '',
+          subject: '',
+          logicalOp: '',
+          _showSubject: true,
+          _showLogicalOp: true,
+          _addedCondition: false
+        };
+        this.items.splice(index, 0, item);
```
Comment (rodriquelca) [inline resources/js/components/PMColumnFilterPopover/PMColumnFilterForm.vue:129]: is possible use push method?

File: resources/js/components/PMColumnFilterPopover/PMColumnFilterForm.vue [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+      },
+      removeItem(index) {
+        this.items.splice(index, 1);
+      },
+      getOperators() {
+        return [
+          {value: '', text: 'Select an option'},
```
Comment (rodriquelca) [inline resources/js/components/PMColumnFilterPopover/PMColumnFilterForm.vue:136]: hardcoded Text, please translate.

File: resources/js/components/shared/FilterTable.vue [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+
+<script>
+
+import moment from "moment";
+
+export default {
+  components: {
+  },
+  props: {
+    headers: [],
```
Comment (rodriquelca) [inline resources/js/components/shared/FilterTable.vue:79]: add prop types and default values
example
` year: {
    default: 2016,
    type: Number
  }`

File: resources/js/components/PMColumnFilterPopover/PMColumnFilterForm.vue [hunk 8]
```diff
@@ -0,0 +1,164 @@
...
+        {{$t('Apply')}}
+      </b-button>
+    </b-form>
+
+  </div>
+</template>
+
+<script>
+  export default {
+    components: {
```
Comment (rodriquelca) [inline resources/js/components/PMColumnFilterPopover/PMColumnFilterForm.vue:67]: remove empty section

File: resources/js/components/shared/FilterTable.vue [hunk 9]
```diff
@@ -0,0 +1,251 @@
...
+      if (this.isResizing) {
+        document.removeEventListener('mousemove', this.doResize);
+        document.removeEventListener('mouseup', this.stopResize);
+        this.isResizing = false;
+        this.resizingColumnIndex = -1;
+      }
+    },
+    formatDate(date, mask) {
+      if (mask === 'datetime') {
+        return date === null ? "-" : moment(date).format("MM/DD/YY HH:mm");
```
Comment (rodriquelca) [inline resources/js/components/shared/FilterTable.vue:151]: hard coded format, maybe you can define as a constant above

File: resources/js/components/shared/FilterTable.vue [hunk 10]
```diff
@@ -0,0 +1,251 @@
...
+      }
+      if (mask === 'date') {
+        return date === null ? "-" : moment(date).format("MM/DD/YY");
```
Comment (rodriquelca) [inline resources/js/components/shared/FilterTable.vue:154]: hard coded format, maybe you can define as a constant above

File: resources/js/components/shared/PmqlInput.vue [hunk 11]
```diff
@@ -64,16 +75,25 @@
...
+              data-bs-toggle="tooltip"
+              class="pmql-badge"
+              id="pmql-pill"
+            >
+              PMQL
+            </b-badge>
           </div>
 
-          <div v-if="showPmqlSection && condensed">
+          <!-- <div v-if="showPmqlSection && condensed">
```
Comment (rodriquelca) [inline resources/js/components/shared/PmqlInput.vue:91]: remove commented code

File: resources/js/components/shared/PmqlInputFilters.vue [hunk 12]
```diff
@@ -1,10 +1,10 @@
 <template>
   <div>
     <div class="position-relative">
-      <button ref="filterButton" class="btn btn-outline-secondary mr-1 d-flex align-items-center" @click="toggleFiltersPopup">
+      <!-- <button ref="filterButton" class="btn btn-outline-secondary mr-1 d-flex align-items-center" @click="toggleFiltersPopup">
```
Comment (rodriquelca) [inline resources/js/components/shared/PmqlInputFilters.vue:4]: remove commented code

File: resources/js/requests/components/RequestsListing.vue [hunk 13]
```diff
@@ -185,9 +159,9 @@ export default {
       });
 
       // this is needed because fields in vuetable2 are not reactive
-      this.$nextTick(() => {
-        this.$refs.vuetable.normalizeFields();
-      });
+      // this.$nextTick(() => {
```
Comment (rodriquelca) [inline resources/js/requests/components/RequestsListing.vue:162]: remove commented code

**Comment by nolanpro** [comment]
@gustavobascope this is looking good! Here are a few things I noticed:
image
Figma
image

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-9d448ab1bf.eng.processmaker.net

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-9d448ab1bf.eng.processmaker.net

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-9d448ab1bf.eng.processmaker.net

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-9d448ab1bf.eng.processmaker.net

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-9d448ab1bf.eng.processmaker.net

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-9d448ab1bf.eng.processmaker.net

**Review by nolanpro — APPROVED** [review]

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-9d448ab1bf.eng.processmaker.net

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-9d448ab1bf.eng.processmaker.net

**Comment by tokensPM** [comment]
QA server was successfully deployed https://ci-9d448ab1bf.eng.processmaker.net

**Pull request merged**